From 84142076f6952e74a7d48388c8f3f2f2ea50218a Mon Sep 17 00:00:00 2001
From: Ashutosh Chauhan <hashutosh@apache.org>
Date: Mon, 2 Jul 2012 08:36:06 +0000
Subject: [PATCH 112/148] HIVE-3207 : FileUtils.tar does not close input files (Navis Ryu via Ashutosh Chauhan)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1356108 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit bf339f73e3fd7fe3df6617ca98044084340106cc)
---
 .../org/apache/hadoop/hive/common/FileUtils.java   |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/common/src/java/org/apache/hadoop/hive/common/FileUtils.java b/src/common/src/java/org/apache/hadoop/hive/common/FileUtils.java
index 010c0ca..096c4ad 100644
--- a/src/common/src/java/org/apache/hadoop/hive/common/FileUtils.java
+++ b/src/common/src/java/org/apache/hadoop/hive/common/FileUtils.java
@@ -267,7 +267,12 @@ public final class FileUtils {
         TarArchiveEntry tarEntry = new TarArchiveEntry(f, f.getName());
         tOut.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
         tOut.putArchiveEntry(tarEntry);
-        IOUtils.copy(new FileInputStream(f), tOut); // copy with 8K buffer, not close
+        FileInputStream input = new FileInputStream(f);
+        try {
+          IOUtils.copy(input, tOut); // copy with 8K buffer, not close
+        } finally {
+          input.close();
+        }
         tOut.closeArchiveEntry();
       }
       tOut.close(); // finishes inside
-- 
1.7.0.4

